src/service-worker/ServiceWorker.ts

```diff
@@ -802,7 +802,14 @@ export class ServiceWorker {
       }
       Log.info("NotificationClicked", notificationClicked);
       saveNotificationClickedPromise = (async (notificationClicked) => {
-        return await Database.put("NotificationClicked", notificationClicked)
+        try {
```

Comment:
TOOD: @jkasten2 Look into if mobile upgrades session when a notification is clicked to direct

Reply:
I checked the Android SDK and confirmed that the session is not changed if the app is not in focus.

It would more correct to detect here if any page is in focus when clicking though. This detection should be done here instead of the `getCurrentSession()` check. However not sure if the fact of clicking on a notification creates any sort of race condition, however the de-bounce logic in the service worker will probably account for this.